use-cases/structured-outputs.mdx

```diff
@@ -0,0 +1,167 @@
+    evaluation.log(
+        "pickup_address_correct",
+        index=idx,
+        passed=pickup_correct,
+        data={
+            "output": extracted_data.pickup_address,
+            "expected": ground_truth.get("pickup_address")
+        }
+    )
+
+    # Check if 'airport_found' flag is correct
+    airport_flag_correct = extracted_data.airport_found == ground_truth.get("airport_found")
+    evaluation.log(
+        "airport_found_correct",
+        index=idx,
+        passed=airport_flag_correct,
+        data={
+            "output": extracted_data.airport_found,
+            "expected": ground_truth.get("airport_found")
+        }
+    )
+    
+    # Check for hallucinations (fields that shouldn't exist)
+    hallucinated_destination = "destination_address" in extracted_data and ground_truth.get("destination_address") is None
+    evaluation.log(
+        "hallucination_check",
+        index=idx,
+        passed=not hallucinated_destination, # Pass if no hallucination
+        data={
+            "output": extracted_data.destination_address
+        }
+    )
+
+    # 3. Log a summary for the entire sample
+    is_fully_correct = pickup_correct and airport_flag_correct and not hallucinated_destination
+    evaluation.log(
+        "overall_correctness",
+        index=idx,
+        passed=is_fully_correct,
+        data={
+            "input": user_message,
+            "output_json": extracted_data,
+            "expected_json": ground_truth,
+        }
+    )
+
+print("Evaluation complete! Check your results in the LangWatch dashboard.")
+```
+
+### **5. Analyzing the Results**
+
+After running the script, you can navigate to the LangWatch dashboard to see your results. You'll get:
+
+* **High-Level Metrics**: An overview of correctness scores across your dataset.
+* **Sample-by-Sample Breakdown**: The ability to inspect each user message, see the model's output vs. the expected output, and identify exactly where it failed.
+* **Historical Tracking**: A record of all your evaluation runs, so you can easily compare model versions and track improvements over time.
+
+For example, you could quickly filter for all samples where `hallucination_check` failed to debug why your model is inventing a destination_address. This level of detail is crucial for iterating on your prompts and improving model reliability.
+
+### **6. Conclusion**
+
+By implementing this evaluation-driven approach with LangWatch, you can systematically measure and improve the accuracy of your structured data extraction for your chatbot. The detailed field-by-field analysis helps identify specific areas for improvement, whether it's handling incomplete addresses, detecting airport mentions, or preventing hallucinations. With continuous monitoring, you can ensure your booking system remains reliable as it processes real-world, unstructured user messages.
+
+For more implementation examples, check out our [RAG cookbook](/cookbooks/build-a-simple-rag-app).
```

Comment:
instead of the rag cookbook, I would actually suggest to add a # 7. Next Steps: Optimization, and link to optimization studio or dspy visualization, because this is exactly the perfect use case for it

later on we should probably write a cookbook for this just optimizing with dspy